vwr/driver.py

```diff
@@ -59,18 +55,18 @@ def _reconnect(self):
         if self.reconnect_trials < 10:
             try:
                 self._connect()
-                self.reconnect_loop.stop()
+                self.get_setpoint()
+                self.reconnect_loop = Timer(self.initialDelay, self._reconnect)
                 self.reconnect_trials = 0
             except:
+                Timer(self.currentDelay, self._reconnect).start()
                 self.reconnect_trials += 1
         else:
             if self.currentDelay < self.maxDelay:
                 self.currentDelay += 20
             else:
                 self.currentDelay = self.maxDelay
```

Comment:
You can cut out all of these lines of code by writing out the exponential decay. I gave you [this source](https://github.com/twisted/twisted/blob/trunk/src/twisted/internet/protocol.py#L332-L464), which has an implementation although may be overly confusing with its use of scientific constants. The important part is:

```python
self.delay = min(self.delay * self.factor, self.maxDelay)
```

And all that matters is that `1 < self.factor < ~5`. The jitter is a nice add, but not needed here.